src/components/OSCALControlProse.test.js

```diff
@@ -59,66 +132,75 @@ function proseParamValuesEditingRenderer() {
 }
 
 function testOSCALControlProseEditing(parentElementName, renderer) {
-  beforeEach(() => {
-    renderer();
-    screen
-      .getByRole("button", {
-        name: "edit-bycomponent-component-1-statement-control-1_smt.a",
-      })
-      .click();
-  });
+  const controlName = "control 1";
+  const componentName = "component 1";
+  const parameterValue = "control-1_prm_1";
+  const statementValue = "save-control-1_smt.a";
+  const rollName = "edit-bycomponent-component-1-statement-control-1_smt.a";
 
   test(`${parentElementName} displays default description and parameter inputs`, async () => {
-    const descriptionTextField = screen.getByLabelText("Description");
-    const paramValueTextField = screen.getByLabelText("control-1_prm_1");
-    expect(descriptionTextField.value).toBe(
-      "Component 1 description of implementing control 1"
-    );
-
-    expect(paramValueTextField.value).toBe(
-      "control 1 / component 1 / parameter 1 value"
-    );
+    renderOSCALControlProseEditing(renderer, rollName);
+    testDefaultDescription(controlName, componentName, parameterValue);
   });
 
   test(`${parentElementName} handles edit with just description edit and keeps placeholders`, async () => {
-    const descriptionTextField = screen.getByLabelText("Description");
-    const paramValueTextField = screen.getByLabelText("control-1_prm_1");
-    fireEvent.change(descriptionTextField, {
-      target: { value: "Test Description" },
-    });
-    screen
-      .getByRole("button", {
-        name: "save-control-1_smt.a",
-      })
-      .click();
-    expect(descriptionTextField.value).toBe("Test Description");
-
-    screen.getByLabelText("control-1_prm_1");
-    expect(paramValueTextField.value).toBe(
-      "control 1 / component 1 / parameter 1 value"
-    );
+    renderOSCALControlProseEditing(renderer, rollName);
```

Comment:
So I do appreciate removing duplication in code, but think that it can be helpful with tests. Please see https://github.com/EasyDynamics/oscal-react-library/pull/525#discussion_r934537769. This duplication was very intentional.

I think in general taking all the code out into various functions makes the tests harder to read and understand.